vendors/st/boards/stm32l475_discovery/ports/common_io/iot_i2c.c

```diff
@@ -0,0 +1,744 @@
+            .Init.GeneralCallMode = I2C_GENERALCALL_DISABLE,
+            .Init.NoStretchMode   = I2C_NOSTRETCH_DISABLE,
+        },
+        .xConfig ={ 0            },
+        .xSemphr = NULL,
+        .usSlaveAddr = _UNSET_SLAVE_ADDRESS,
+        .usTransmittedTxBytes = 0,
+        .usReceivedRxBytes = 0,
+        .flags = _FLAG_INITIALIZER
+    },
+
+    {
+        .xHandle =
+        {
+            .Instance             = I2C3,
+            .Init.Timing          = DISCOVERY_I2Cx_TIMING,
+            .Init.OwnAddress1     = 0,
+            .Init.AddressingMode  = I2C_ADDRESSINGMODE_7BIT,
+            .Init.DualAddressMode = I2C_DUALADDRESS_DISABLE,
+            .Init.OwnAddress2     = 0,
+            .Init.GeneralCallMode = I2C_GENERALCALL_DISABLE,
+            .Init.NoStretchMode   = I2C_NOSTRETCH_DISABLE,
+        },
+        .xConfig ={ 0            },
+        .xSemphr = NULL,
+        .usSlaveAddr = _UNSET_SLAVE_ADDRESS,
+        .usTransmittedTxBytes = 0,
+        .usReceivedRxBytes = 0,
+        .flags = _FLAG_INITIALIZER
+    }
+};
+
+
+
+/*---------------------Private functions---------------------*/
+
+static int32_t _toI2cStatus( HAL_StatusTypeDef halStatus,
+                             uint8_t operation );
+
+static int32_t _doMasterTransfer( IotI2CDescriptor_t * pI2cDescriptor,
+                                  uint8_t * const pucBuffer,
+                                  size_t xBytes,
+                                  HAL_StatusTypeDef ( * pMasterSync )( I2C_HandleTypeDef * hi2c,
+                                                                       uint16_t DevAddress,
+                                                                       uint8_t * pData,
+                                                                       uint16_t Size,
+                                                                       uint32_t Timeout ),
+                                  HAL_StatusTypeDef ( * pMasterIT )( I2C_HandleTypeDef * hi2c,
+                                                                     uint16_t DevAddress,
+                                                                     uint8_t * pData,
+                                                                     uint16_t Size ),
+                                  HAL_StatusTypeDef ( * pMasterSequentialIT )( I2C_HandleTypeDef * hi2c,
+                                                                               uint16_t DevAddress,
+                                                                               uint8_t * pData,
+                                                                               uint16_t Size,
+                                                                               uint32_t XferOptions ) );
+
+/*--------------------API Implementation---------------------*/
+
+IotI2CHandle_t iot_i2c_open( int32_t lI2CInstance )
+{
+    /* Exceeded maximum instance number. */
+    if( lI2CInstance >= _I2C_INSTANCE_NUM )
```

Comment:
Should this also check if lI2CInstance < 0, i seem to remember this used to take an unsigned int, but should be updated now that it is signed

Reply:
Good catch. I should check <0.

Reply:
This is what I am referencing in the UART open comment, it seems your answer there indicates that this should then be <, not <=